core/modules/replica/CreateReplicaJob.cc

```diff
@@ -201,11 +204,14 @@ void CreateReplicaJob::startImpl(util::Lock const& lock) {
     //    see if the chunk is available on a source node.
 
     std::vector<ReplicaInfo> sourceReplicas;
-    if (not controller()->serviceProvider()->databaseServices()->findWorkerReplicas(
-                sourceReplicas,
-                chunk(),
-                sourceWorker(),
-                databaseFamily())) {
+    try {
+        controller()->serviceProvider()->databaseServices()->findWorkerReplicas(
+            sourceReplicas,
+            chunk(),
+            sourceWorker(),
+            databaseFamily());
+
+    } catch (std::exception const&) {
```

Comment:
there's not a more specific exception type that would get thrown/should get caught? `std::exception` is kind of a big umbrella.

Reply:
This code is calling a method (via the database-neural DatabaseServices interface) which attempts to pull replicas (for a given set of parameters) from a database. Implementations of this method may vary depending on a database technology used in there. I intentionally don't want to depend on database-specific exceptions in this code. If I see any of those I just report a problem and graceful shut down the replica creation job because those error conditions may be of a transient nature (like, MySQL.MariaDB service is temporarily down, etc.). However, there is one type of exceptions which can still be recognized here:
`std::logic_error`
it's thrown when something is wrong with the selection criteria passed into the replica lookup method. Typically this happens if the application is misconfigured. Hence, I'm going to reinforce the code to recognize this particular situation, report it and re-throw the exception (which will result in terminating the application, and which is the right thing to do here).